### ISSUE TITLE
TypeError: cli_utils_1.generateIonicEnvironment is not a function

### ISSUE THREAD

**Issue by janpio** [ionic-team/ionic-cli#2605]
I just wanted to build an old project, that was last built a month ago:

```
PS C:\Users\Jan\Documents\ionic-android-contacts> ionic cordova run android
[WARN] No local CLI detected.
       Starting with CLI 3.6, the CLI must be installed locally to use local CLI plugins.

? Install now? Yes
> npm install --save-dev --save-exact ionic@latest
√ Running command - done!
[OK] Installed Ionic CLI locally!

Please re-run your command.
```

But now I get this:

```
PS C:\Users\Jan\Documents\ionic-android-contacts> ionic cordova run android
TypeError: cli_utils_1.generateIonicEnvironment is not a function
    at Object.<anonymous> (C:\Users\Jan\Documents\ionic-android-contacts\node_modules\ionic\dist\index.js:84:40)
    at Generator.next (<anonymous>)
    at C:\Users\Jan\Documents\ionic-android-contacts\node_modules\tslib\tslib.js:96:75
    at Promise (<anonymous>)
    at Object.__awaiter (C:\Users\Jan\Documents\ionic-android-contacts\node_modules\tslib\tslib.js:92:16)
    at Object.run (C:\Users\Jan\Documents\ionic-android-contacts\node_modules\ionic\dist\index.js:78:20)
    at C:\nvm\v8.1.3\node_modules\ionic\bin\ionic:21:7
    at <anonymous>
```

For all `ionic` commands:

```
PS C:\Users\Jan\Documents\ionic-android-contacts> ionic info
TypeError: cli_utils_1.generateIonicEnvironment is not a function
    at Object.<anonymous> (C:\Users\Jan\Documents\ionic-android-contacts\node_modules\ionic\dist\index.js:84:40)
    at Generator.next (<anonymous>)
    at C:\Users\Jan\Documents\ionic-android-contacts\node_modules\tslib\tslib.js:96:75
    at Promise (<anonymous>)
    at Object.__awaiter (C:\Users\Jan\Documents\ionic-android-contacts\node_modules\tslib\tslib.js:92:16)
    at Object.run (C:\Users\Jan\Documents\ionic-android-contacts\node_modules\ionic\dist\index.js:78:20)
    at C:\nvm\v8.1.3\node_modules\ionic\bin\ionic:21:7
    at <anonymous>
```

```
PS C:\Users\Jan\Documents\ionic-android-contacts> ionic -v
TypeError: cli_utils_1.generateIonicEnvironment is not a function
    at Object.<anonymous> (C:\Users\Jan\Documents\ionic-android-contacts\node_modules\ionic\dist\index.js:84:40)
    at Generator.next (<anonymous>)
    at C:\Users\Jan\Documents\ionic-android-contacts\node_modules\tslib\tslib.js:96:75
    at Promise (<anonymous>)
    at Object.__awaiter (C:\Users\Jan\Documents\ionic-android-contacts\node_modules\tslib\tslib.js:92:16)
    at Object.run (C:\Users\Jan\Documents\ionic-android-contacts\node_modules\ionic\dist\index.js:78:20)
    at C:\nvm\v8.1.3\node_modules\ionic\bin\ionic:21:7
    at <anonymous>
```

Possibly because it installed a local CLI 3.7 and the global one is still on 3.6?

```
PS C:\Users\Jan\Documents\ionic-android-contacts> cd ..
PS C:\Users\Jan\Documents> ionic -v
? The Ionic CLI has an update available (3.6.0 => 3.7.0)! Would you like to install it? No
[INFO] Not automatically updating your CLI. You can update manually:

       npm install -g ionic@latest

3.6.0
PS C:\Users\Jan\Documents>
```

Snippet of `package.json` of that project:
```
        "@ionic/app-scripts": "1.3.12",
        "@ionic/cli-plugin-cordova": "1.4.0",
        "@ionic/cli-plugin-ionic-angular": "1.3.1",
```

**Comment by janpio** [comment]
Updating the global install to 3.7 didn't fix the problem.

**Comment by janpio** [comment]
```
PS C:\Users\Jan\Documents\ionic-android-contacts> npm list --depth=0
ionic-android-contacts@0.0.1 C:\Users\Jan\Documents\ionic-android-contacts
+-- @angular/common@4.1.3
+-- @angular/compiler@4.1.3
+-- @angular/compiler-cli@4.1.3
+-- @angular/core@4.1.3
+-- @angular/forms@4.1.3
+-- @angular/http@4.1.3
+-- @angular/platform-browser@4.1.3
+-- @angular/platform-browser-dynamic@4.1.3
+-- @ionic-native/contacts@3.13.1
+-- @ionic-native/core@3.12.1
+-- @ionic-native/splash-screen@3.12.1
+-- @ionic-native/status-bar@3.12.1
+-- @ionic/app-scripts@1.3.12
+-- @ionic/cli-plugin-cordova@1.4.0
+-- @ionic/cli-plugin-ionic-angular@1.3.1
+-- @ionic/storage@2.0.1
+-- cordova-android@6.2.3
+-- cordova-plugin-compat@1.0.0
+-- cordova-plugin-console@1.0.5
+-- cordova-plugin-contacts@2.3.1
+-- cordova-plugin-device@1.1.4
+-- cordova-plugin-splashscreen@4.0.3
+-- cordova-plugin-statusbar@2.2.2
+-- cordova-plugin-whitelist@1.3.1
+-- ionic@3.7.0
+-- ionic-angular@3.5.0
+-- ionic-plugin-keyboard@2.2.1
+-- ionicons@3.0.0
+-- rxjs@5.4.0
+-- sw-toolbox@3.6.0
+-- typescript@2.3.4
`-- zone.js@0.8.12
```

**Comment by janpio** [comment]
I fixed it for now by doing this:

```
PS C:\Users\Jan\Documents\ionic-android-contacts> npm install --save-dev @ionic/cli-utils
npm WARN @ionic/cli-utils@1.4.0 requires a peer of strip-ansi@^3.0.1 but none was installed.
npm WARN @ionic/cli-utils@1.4.0 requires a peer of strip-ansi@^3.0.1 but none was installed.

+ @ionic/cli-utils@1.7.0
added 25 packages, updated 2 packages and moved 1 package in 16.585s
```

```
PS C:\Users\Jan\Documents\ionic-android-contacts> ionic info
? Local plugin @ionic/cli-plugin-cordova has an update available (1.4.0 => 1.6.1)! Would you like to install it? Yes
> npm install --save-dev --save-exact @ionic/cli-plugin-cordova@latest
√ Running command - done!
[OK] Updated @ionic/cli-plugin-cordova to 1.6.1! 🎉
? Local plugin @ionic/cli-plugin-ionic-angular has an update available (1.3.1 => 1.4.1)! Would you like to install it?
Yes
> npm install --save-dev --save-exact @ionic/cli-plugin-ionic-angular@latest
√ Running command - done!
[OK] Updated @ionic/cli-plugin-ionic-angular to 1.4.1! 🎉
> npm dedupe
√ Running command - done!

cli packages: (C:\Users\Jan\Documents\ionic-android-contacts\node_modules)

    @ionic/cli-plugin-cordova       : 1.6.1
    @ionic/cli-plugin-ionic-angular : 1.4.1
    @ionic/cli-utils                : 1.7.0
    ionic (Ionic CLI)               : 3.7.0

global packages:

    Cordova CLI : 7.0.1

local packages:

    @ionic/app-scripts : 1.3.12
    Cordova Platforms  : android 6.2.3
    Ionic Framework    : ionic-angular 3.5.0

System:

    Android SDK Tools : 26.0.2
    Node              : v8.1.3
    OS                : Windows 10
    npm               : 5.1.0

```

So my guess: old, locally installed cli-utils were not compatible with CLI 3.7 and made `ionic` fail before the update checks.

**Comment by dwieeb** [comment]
Hey all, 3.8 no longer loads the CLI plugins, so this error won't happen anymore.
